caddyhttp/proxy/reverseproxy.go

```diff
@@ -331,51 +340,148 @@ type connHijackerTransport struct {
 }
 
 func newConnHijackerTransport(base http.RoundTripper) *connHijackerTransport {
-	transport := &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
-		Dial: (&net.Dialer{
-			Timeout:   30 * time.Second,
-			KeepAlive: 30 * time.Second,
-		}).Dial,
-		TLSHandshakeTimeout: 10 * time.Second,
+	t := &http.Transport{
 		MaxIdleConnsPerHost: -1,
 	}
-	if base != nil {
-		if baseTransport, ok := base.(*http.Transport); ok {
-			transport.Proxy = baseTransport.Proxy
-			transport.TLSClientConfig = baseTransport.TLSClientConfig
-			transport.TLSHandshakeTimeout = baseTransport.TLSHandshakeTimeout
-			transport.Dial = baseTransport.Dial
-			transport.DialTLS = baseTransport.DialTLS
-			transport.MaxIdleConnsPerHost = -1
+	if b, _ := base.(*http.Transport); b != nil {
```

Comment:
Nice catch! If `ok` is false (and the cast thus failed) `b` must be `nil` right? (As per Go spec.)

So if we have this code:
```go
if base != nil {
    if b, ok := base.(*http.Transport); ok == true {
        ...
    }
}
```

It's the same as this:
```go
if b, _ := base.(*http.Transport); b != nil {
}
```

because if `base` is `nil` it stays `nil` and if it's not a `*http.Transport` it will be converted to `nil`. I usually use this coding pattern in Go if I want to write sth. along the former snippet. Should I revert that or add a comment?